src/Interpreters/InterpreterDeleteQuery.cpp

```diff
@@ -0,0 +1,89 @@
+#include <Interpreters/InterpreterDeleteQuery.h>
+
+#include <Access/ContextAccess.h>
+#include <Databases/DatabaseReplicated.h>
+#include <Databases/IDatabase.h>
+#include <Interpreters/Context.h>
+#include <Interpreters/FunctionNameNormalizer.h>
+#include <Interpreters/MutationsInterpreter.h>
+#include <Parsers/ASTDeleteQuery.h>
+#include <Storages/AlterCommands.h>
+#include <Storages/IStorage.h>
+#include <Storages/MutationCommands.h>
+#include <Storages/StorageMergeTree.h>
+
+
+namespace DB
+{
+
+namespace ErrorCodes
+{
+    extern const int BAD_ARGUMENTS;
+    extern const int TABLE_IS_READ_ONLY;
+}
+
+
+InterpreterDeleteQuery::InterpreterDeleteQuery(const ASTPtr & query_ptr_, ContextPtr context_) : WithContext(context_), query_ptr(query_ptr_)
+{
+}
+
+
+BlockIO InterpreterDeleteQuery::execute()
+{
+    FunctionNameNormalizer().visit(query_ptr.get());
```

Comment:
Let's introduce a setting like `allow_experimental_lightweight_delete` which is disabled by default.